src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -247,4 +251,70 @@ tsi_result SslProtectorUnprotect(const unsigned char* protected_frames_bytes,
   return result;
 }
 
+bool VerifyCrlSignature(X509_CRL* crl, X509* issuer) {
+  if (issuer == nullptr || crl == nullptr) {
+    return false;
+  }
+  EVP_PKEY* ikey = X509_get_pubkey(issuer);
+  if (ikey == nullptr) {
+    // Can't verify signature because we couldn't get the pubkey, fail the
+    // check.
+    EVP_PKEY_free(ikey);
+    return false;
+  }
+  bool ret = X509_CRL_verify(crl, ikey) == 1;
+  EVP_PKEY_free(ikey);
+  return ret;
+}
+
+bool VerifyCrlCertIssuerNamesMatch(X509_CRL* crl, X509* cert) {
+  if (cert == nullptr || crl == nullptr) {
+    return false;
+  }
+  X509_NAME* cert_issuer_name = X509_get_issuer_name(cert);
+  if (cert == nullptr) {
+    return false;
+  }
+  X509_NAME* crl_issuer_name = X509_CRL_get_issuer(crl);
+  if (crl_issuer_name == nullptr) {
+    return false;
+  }
+  return X509_NAME_cmp(cert_issuer_name, crl_issuer_name) == 0;
+}
+
+bool HasCrlSignBit(X509* cert) {
+  if (cert == nullptr) {
+    return false;
+  }
+  // X509_get_key_usage was introduced in 1.1.1
+  // A missing key usage extension means all key usages are valid.
+#if OPENSSL_VERSION_NUMBER < 0x10100000
+  // X509_check_ca seems to have the side effect of setting cert->ex_flags in
```

Comment:
Sure, would a code link suffice? Here we can see the important call that it makes
https://github.com/openssl/openssl/blob/e818b74be2170fbe957a07b0da4401c2b694b3b8/crypto/x509v3/v3_purp.c#L585

Reply:
A code link is fine. But can we also clean up the language so that it is more definite? From your deep dive into OpenSSL 1.0.2, we know the issue, so I just wanted the comment to be phrased in more definite language (e.g. s/seems to have/has)

Reply:
Cleaned up